hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java

```diff
@@ -161,6 +157,8 @@ public void start(Object service) {
     OzoneConfiguration.activate();
     if (service instanceof Configurable) {
       conf = new OzoneConfiguration(((Configurable) service).getConf());
+    } else {
```

Comment:
I think this part can be simplified (but fix me if I am wrong) with using two start method:

```
 /**
   * Starts HddsDatanode services.
   *
   * @param service The service instance invoking this method
   */
  @Override
  public void start(Object service) {
    if (service instanceof Configurable) {
      start(new OzoneConfiguration(((Configurable) service).getConf()));
    } else {
      start(new OzoneConfiguration());
    }
  }

  public void start(OzoneConfiguration conf) {
    this.conf = conf;
    DefaultMetricsSystem.initialize("HddsDatanode");
    OzoneConfiguration.activate();
    if (HddsUtils.isHddsEnabled(conf)) {
 ```

In this case we don't need to add the additional method to the GenericParentCommand so it can be as simple as now.

And from the new call() method we can use the createOzoneConfiguration()

```
  @Override
  public Void call() throws Exception {
    if (printBanner) {
      StringUtils
          .startupShutdownMessage(HddsDatanodeService.class, args, LOG);
    }
    start(createOzoneConfiguration());
    join();
    return null;
  }
```

initConf also can be removed with this approach (IMHO)

Reply:
This is a good idea, I simplified this part, but I introduced a setConfiguration method here as well so that I don't have to modify MiniOzoneCluster that much.